source/grid/grid_generator.cc

```diff
@@ -3625,6 +3625,25 @@ namespace GridGenerator
     GridGenerator::merge_triangulations(
       tria_without_cylinder, cylinder_tria, tria, vertex_tolerance, true);
 
+    // Move the vertices in the middle of the faces of cylinder_tria slightly
+    // to give a better mesh quality. We choose to move them such that the
+    // distance between (0.05 + shell_region_width) and 0.1 is at least half
+    // as long as the remaining length to the boundary.
+    const double shifted_point =
+      std::min(0.05 + shell_region_width + 0.1 * std::sqrt(2) - 0.05 -
+                 shell_region_width,
+               0.1 * 4. / 3.);
```

Comment:
I don't understand the point of this calculation. The first expression is just `0.1*sqrt(2)` so its just a comparison of two constants.

As far as I can tell all that this (and the code below) does is (since `0.1*4.0/3.0 < 0.1*sqrt(2)`) nudge the vertices out by `0.1/3.0` in the x direction and `0.1/3.0 - 0.0025` (or `0.1/3.0 - 0.0075`) in the y direction. Is this right? Did you mean for it to rely on the shell region width?

Reply:
You are absolutely right - I missed the forest for the trees. The formula comes from how I constructed a better shape in radial distance by some angles I had drawn on paper. In the end, this turned out to just be a constant distance from the center in radial direction (`0.1 * sqrt(2)`). And you are right, once one realizes that things turn out constant, the `0.1 * 4./3.` is the smaller number of the two (making sure the outer elements are not too distorted), so we can simply put that number. I will add this describing text next to the place in the code so we remember how we arrived here.